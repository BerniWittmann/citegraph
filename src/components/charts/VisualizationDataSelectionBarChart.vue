<script lang="ts">
import { Bar } from 'vue-chartjs'
import { Component, Prop, Mixins, Watch } from 'vue-property-decorator'
import { materialColors } from '@/common/constants'
import hexToRgba from 'hex-to-rgba'

interface ChartData {
  x: number,
  y: number
}

@Component
export default class VisualizationDataSelectionBarChart extends Mixins(Bar) {
  @Prop() readonly entityData!: Record<string, number>
  @Prop({ default: 'Data' }) readonly entityType!: string
  @Prop() readonly timePeriods!: Array<any>

  colors: Array<string> = materialColors.sort(() => 0.5 - Math.random()) // Randomly sort colors

  get barData (): Array<ChartData> {
    const result = []
    for (let year in this.entityData) {
      result.push({
        x: parseInt(year),
        y: this.entityData[year]
      })
    }
    return result
  }

  get timePeriodsDataSets (): Array<any> {
    return this.timePeriods.map((timePeriod, i) => {
      if (!timePeriod.min || !timePeriod.max) return undefined
      const min = parseInt(timePeriod.min)
      const max = parseInt(timePeriod.max)
      if (min > max) return undefined
      const color = this.colors[i % this.colors.length]
      return {
        data: [{
          x: min + '-01-01',
          y: this.maxValue
        }, {
          x: max + '-01-01',
          y: this.maxValue
        }],
        fill: 'origin',
        type: 'line',
        backgroundColor: hexToRgba(color, 0.1),
        pointRadius: 0,
        borderWidth: 0,
        label: timePeriod.id
      }
    }).filter((v: any) => !!v)
  }

  get chartData (): Object {
    if (this.barData.length === 0) {
      return {
        datasets: []
      }
    }
    return {
      datasets: [{
        data: this.barData.map(d => ({
          ...d,
          x: d.x + '-01-01'
        })),
        label: this.entityType
      }, ...this.timePeriodsDataSets]
    }
  }

  get minYear (): number {
    if (this.barData.length === 0) {
      return 1971
    }
    return this.barData.reduce((acc: number, val: ChartData) => {
      return val.x < acc ? val.x : acc
    }, Number.MAX_SAFE_INTEGER)
  }

  get maxValue (): number {
    return this.barData.reduce((acc: number, val: ChartData) => {
      return val.y > acc ? val.y : acc
    }, 0)
  }

  get maxYear (): number {
    if (this.barData.length === 0) {
      return (new Date()).getFullYear()
    }

    return this.barData.reduce((acc: number, val: ChartData) => {
      if (val.x > acc) {
        return val.x
      } else {
        return acc
      }
    }, 0)
  }

  get chartOptions (): Object {
    return {
      responsive: true,
      maintainAspectRatio: false,
      scales: {
        xAxes: [{
          type: 'time',
          bounds: 'data',
          gridLines: {
            display: true,
            offsetGridLines: true
          },
          time: {
            unit: 'year',
            tooltipFormat: 'YYYY'
          },
          ticks: {
            min: (this.minYear - 1) + '-01-01',
            max: (this.maxYear + 1) + '-01-01'
          }
        }],
        yAxes: [{
          ticks: {
            beginAtZero: true,
            precision: 0
          }
        }]
      },
      legend: {
        display: false
      }
    }
  }

  @Watch('chartData', { deep: true })
  updateChart () {
    this.renderChart(this.chartData, this.chartOptions)
  }

  mounted (): void {
    this.updateChart()
  }
}
</script>
