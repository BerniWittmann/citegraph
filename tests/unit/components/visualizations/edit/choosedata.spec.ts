import { shallowMount } from '@vue/test-utils'
import { i18n } from '../../../setupPlugins'

import VisualizationEditChooseDataComponent from '@/components/visualizations/edit/ChooseData.vue'
import VisualizationDataSelectionBarChart from '@/components/charts/VisualizationDataSelectionBarChart.vue'

const store = {
  getters: {
    'projects/activeProject': { id: 41 },
    'paperEntities/entities': [
      { id: 1, name: 'A', year: 2011 },
      { id: 2, name: 'B', year: 2010 },
      { id: 3, name: 'C', year: 2011 },
      { id: 4, name: 'invalid' }
    ]
  },
  dispatch: jest.fn()
}

describe('components/visualizations/edit/ChooseData.vue', () => {
  const getWrapper = () => {
    return shallowMount(VisualizationEditChooseDataComponent, {
      i18n,
      mocks: {
        $t: (key: string) => key,
        $store: store
      },
      propsData: {
        name: ''
      }
    })
  }

  it('renders', () => {
    const wrapper = getWrapper()
    // unset Rules to ensure Snapshot stability
    // @ts-ignore
    wrapper.vm.yearRules = []
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders the chart', () => {
    const wrapper = getWrapper()

    const chart = wrapper.find(VisualizationDataSelectionBarChart)
    expect(chart.exists()).toBeTruthy()
  })

  it('has a back Button', () => {
    const wrapper = getWrapper()
    const btns = wrapper.findAll('v-btn-stub')
    const btn = btns.at(btns.length - 2)
    expect(btn.exists()).toBeTruthy()
    expect(btn.text()).toContain('visualizations.add.data.back')
    expect(wrapper.emitted('previous-step')).toBeFalsy()
    btn.vm.$emit('click')
    expect(wrapper.emitted('previous-step')).toBeTruthy()
  })

  describe('it has a text field of the year', () => {
    it('the year needs to be entered', () => {
      const wrapper = getWrapper()
      // @ts-ignore
      const rules = wrapper.vm.yearRules
      expect.assertions(rules.length)
      for (let rule of rules) {
        expect(rule('2019')).toBeTruthy()
      }
    })
    it('the text is required', () => {
      const wrapper = getWrapper()
      // @ts-ignore
      const rule = wrapper.vm.yearRules[0]
      expect(rule(undefined)).toEqual('visualizations.add.data.period.year_required')
      expect(rule(null)).toEqual('visualizations.add.data.period.year_required')
      expect(rule('')).toEqual('visualizations.add.data.period.year_required')
    })
    it('the text can not be a too low number', () => {
      const wrapper = getWrapper()
      // @ts-ignore
      const rule = wrapper.vm.yearRules[1]
      expect(rule(1)).toEqual('visualizations.add.data.period.year_min')
      expect(rule(1970)).toBeTruthy()
      expect(rule(1969)).toEqual('visualizations.add.data.period.year_min')
    })
    it('the text can not be a too high number', () => {
      const wrapper = getWrapper()
      // @ts-ignore
      const rule = wrapper.vm.yearRules[2]
      expect(rule(819358915)).toEqual('visualizations.add.data.period.year_max')
      expect(rule((new Date()).getFullYear())).toBeTruthy()
      expect(rule((new Date()).getFullYear() + 1)).toEqual('visualizations.add.data.period.year_max')
    })
  })

  describe('it can go to the next step', () => {
    it('has a submit button', () => {
      const wrapper = getWrapper()
      const btns = wrapper.findAll('v-btn-stub')
      const btn = btns.at(btns.length - 1)
      expect(btn.exists()).toBeTruthy()
      expect(btn.text()).toContain('visualizations.add.data.next')
    })

    it('emits the next step event on click', () => {
      const wrapper = getWrapper()
      const btns = wrapper.findAll('v-btn-stub')
      const btn = btns.at(btns.length - 1)
      expect(wrapper.emitted('next-step')).toBeFalsy()
      btn.vm.$emit('click')
      expect(wrapper.emitted('next-step')).toBeTruthy()
    })
  })

  describe('loads the data', () => {
    it('loads the data on mount', () => {
      getWrapper()
      expect(store.dispatch).toHaveBeenCalledWith('paperEntities/fetchEntities', {
        projectId: 41,
        entityType: 'record'
      })
    })

    it('reloads the data on type change', () => {
      const wrapper = getWrapper()
      jest.clearAllMocks()
      // @ts-ignore
      wrapper.vm.loadData()
      expect(store.dispatch).toHaveBeenCalledWith('paperEntities/fetchEntities', {
        projectId: 41,
        entityType: 'record'
      })
    })

    it('does not load the data on type the can not select by year', () => {
      const wrapper = getWrapper()
      jest.clearAllMocks()
      // @ts-ignore
      wrapper.vm.dataType = 'author'
      // @ts-ignore
      wrapper.vm.loadData()
      expect(store.dispatch).not.toHaveBeenCalled()
    })
  })

  describe('it handles the time periods', () => {
    it('can add a time period', () => {
      const wrapper = getWrapper()
      const btn = wrapper.find('v-btn-stub')
      expect(btn.exists()).toBeTruthy()
      expect(btn.text()).toEqual('visualizations.add.data.period.add')
      btn.vm.$emit('click')
      // unset Rules to ensure Snapshot stability
      // @ts-ignore
      wrapper.vm.yearRules = []
      expect(wrapper.html()).toMatchSnapshot()
      expect(wrapper.contains('v-text-field-stub')).toBeTruthy()
    })

    it('can remove a time period', () => {
      const wrapper = getWrapper()
      const btn = wrapper.find('v-btn-stub')
      btn.vm.$emit('click')
      // unset Rules to ensure Snapshot stability
      // @ts-ignore
      wrapper.vm.yearRules = []
      expect(wrapper.html()).toMatchSnapshot()
      expect(wrapper.contains('v-text-field-stub')).toBeTruthy()

      const closeBtn = wrapper.findAll('v-btn-stub').at(0)
      expect(closeBtn.exists()).toBeTruthy()
      expect(closeBtn.text()).toEqual('mdi-trash-can-outline')
      closeBtn.vm.$emit('click')

      expect(wrapper.html()).toMatchSnapshot()
      expect(wrapper.contains('v-text-field-stub')).toBeFalsy()
    })
  })

  describe('it correctly passes the time periods', () => {
    it('handles invalid time periods', () => {
      const wrapper = getWrapper()
      const data = [{
        id: 1,
        count: -1,
        min: undefined,
        max: undefined
      }, {
        id: 2,
        count: -1,
        min: 2019,
        max: 2000
      }]
      // @ts-ignore
      wrapper.vm.timePeriodData = data
      expect(wrapper.find(VisualizationDataSelectionBarChart).props('timePeriods')).toEqual(data)
    })

    it('calculates the amount', () => {
      const wrapper = getWrapper()
      const data = [
        { id: 1, count: -1, min: 2000, max: 2015 },
        { id: 2, count: -1, min: 1990, max: 1991 },
        { id: 3, count: -1, min: 2020, max: 2025 }
      ]
      // @ts-ignore
      wrapper.vm.timePeriodData = data
      // @ts-ignore
      expect(wrapper.vm.timePeriods).toEqual([{
        ...data[0],
        count: 3
      }, {
        ...data[1],
        count: 0
      }, {
        ...data[2],
        count: 0
      }])
    })
  })

  it('has a function to update the chart', () => {
    const wrapper = getWrapper()
    // @ts-ignore
    wrapper.vm.$refs.chart.updateChart = jest.fn()
    // @ts-ignore
    expect(wrapper.vm.updateChart).toEqual(expect.any(Function))
    // @ts-ignore
    expect(wrapper.vm.$refs.chart.updateChart).not.toHaveBeenCalled()
    // @ts-ignore
    wrapper.vm.updateChart()
    // @ts-ignore
    expect(wrapper.vm.$refs.chart.updateChart).toHaveBeenCalled()
  })
})
