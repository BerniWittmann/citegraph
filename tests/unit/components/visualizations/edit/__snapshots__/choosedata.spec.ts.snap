// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/visualizations/edit/ChooseData.vue it handles the time periods can add a time period 1`] = `
<div class="visualization-add-data" name="">
  <h1>visualizations.add.data.title</h1>
  <v-container-stub tag="div" fluid="true">
    <v-row-stub tag="div">
      <v-col-stub cols="12" lg="2" tag="div">
        <h4>visualizations.add.data.choose_type</h4>
        <v-list-stub tag="div" tile="true">
          <v-list-item-group-stub value="record" activeclass="v-item--active" mandatory="true" color="primary">
            <v-list-item-stub activeclass="" tag="div" value="record">
              <v-list-item-content-stub>
                <v-list-item-title-stub>
                  visualizations.add.data.data_types.record
                </v-list-item-title-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
            <v-list-item-stub activeclass="" tag="div" value="author">
              <v-list-item-content-stub>
                <v-list-item-title-stub>
                  visualizations.add.data.data_types.author
                </v-list-item-title-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
            <v-list-item-stub activeclass="" tag="div" value="institution">
              <v-list-item-content-stub>
                <v-list-item-title-stub>
                  visualizations.add.data.data_types.institution
                </v-list-item-title-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
            <v-list-item-stub activeclass="" tag="div" value="country">
              <v-list-item-content-stub>
                <v-list-item-title-stub>
                  visualizations.add.data.data_types.country
                </v-list-item-title-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
            <v-list-item-stub activeclass="" tag="div" value="publication">
              <v-list-item-content-stub>
                <v-list-item-title-stub>
                  visualizations.add.data.data_types.publication
                </v-list-item-title-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
            <v-list-item-stub activeclass="" tag="div" value="publisher">
              <v-list-item-content-stub>
                <v-list-item-title-stub>
                  visualizations.add.data.data_types.publisher
                </v-list-item-title-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
            <v-list-item-stub activeclass="" tag="div" value="conference">
              <v-list-item-content-stub>
                <v-list-item-title-stub>
                  visualizations.add.data.data_types.conference
                </v-list-item-title-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
          </v-list-item-group-stub>
        </v-list-stub>
      </v-col-stub>
      <v-col-stub cols="12" lg="7" tag="div">
        <visualization-data-selection-bar-chart-stub chartid="bar-chart" width="400" height="400" cssclasses="" plugins="" entitydata="[object Object]" entitytype="visualizations.add.data.data_types.record" timeperiods=""></visualization-data-selection-bar-chart-stub>
      </v-col-stub>
      <v-col-stub cols="12" lg="3" tag="div">
        <h4>visualizations.add.data.set_time_period</h4>
        <v-container-stub tag="div" fluid="true">
          <v-row-stub tag="div" align="center">
            <v-col-stub cols="3" tag="div">
              <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" backgroundcolor="" label="visualizations.add.data.period.from" loaderheight="2" clearicon="$clear" placeholder="visualizations.add.data.period.enter_year" type="number" min="1970" max="2019"></v-text-field-stub>
            </v-col-stub>
            <v-col-stub cols="3" tag="div">
              <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" backgroundcolor="" label="visualizations.add.data.period.until" loaderheight="2" clearicon="$clear" placeholder="visualizations.add.data.period.enter_year" type="number" min="1970" max="2019"></v-text-field-stub>
            </v-col-stub>
            <v-col-stub cols="auto" tag="div">
              visualizations.add.data.period.elements
            </v-col-stub>
            <v-col-stub cols="auto" tag="div">
              <v-btn-stub tag="button" activeclass="" icon="true" type="button">
                <v-icon-stub>mdi-trash-can-outline</v-icon-stub>
              </v-btn-stub>
            </v-col-stub>
          </v-row-stub>
          <v-row-stub tag="div">
            <v-col-stub tag="div">
              <v-btn-stub tag="button" activeclass="" type="button">visualizations.add.data.period.add</v-btn-stub>
            </v-col-stub>
          </v-row-stub>
        </v-container-stub>
      </v-col-stub>
    </v-row-stub>
    <v-container-stub tag="div">
      <v-row-stub tag="div" justify="space-between" class="my-10">
        <v-col-stub cols="auto" tag="div">
          <v-btn-stub color="warning" tag="button" activeclass="" large="true" rounded="true" type="button">
            <v-icon-stub>mdi-menu-left</v-icon-stub>
            visualizations.add.data.back
          </v-btn-stub>
        </v-col-stub>
        <v-col-stub cols="auto" tag="div">
          <v-btn-stub color="success" tag="button" activeclass="" large="true" rounded="true" type="submit">visualizations.add.data.next
          </v-btn-stub>
        </v-col-stub>
      </v-row-stub>
    </v-container-stub>
  </v-container-stub>
</div>
`;

exports[`components/visualizations/edit/ChooseData.vue it handles the time periods can remove a time period 1`] = `
<div class="visualization-add-data" name="">
  <h1>visualizations.add.data.title</h1>
  <v-container-stub tag="div" fluid="true">
    <v-row-stub tag="div">
      <v-col-stub cols="12" lg="2" tag="div">
        <h4>visualizations.add.data.choose_type</h4>
        <v-list-stub tag="div" tile="true">
          <v-list-item-group-stub value="record" activeclass="v-item--active" mandatory="true" color="primary">
            <v-list-item-stub activeclass="" tag="div" value="record">
              <v-list-item-content-stub>
                <v-list-item-title-stub>
                  visualizations.add.data.data_types.record
                </v-list-item-title-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
            <v-list-item-stub activeclass="" tag="div" value="author">
              <v-list-item-content-stub>
                <v-list-item-title-stub>
                  visualizations.add.data.data_types.author
                </v-list-item-title-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
            <v-list-item-stub activeclass="" tag="div" value="institution">
              <v-list-item-content-stub>
                <v-list-item-title-stub>
                  visualizations.add.data.data_types.institution
                </v-list-item-title-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
            <v-list-item-stub activeclass="" tag="div" value="country">
              <v-list-item-content-stub>
                <v-list-item-title-stub>
                  visualizations.add.data.data_types.country
                </v-list-item-title-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
            <v-list-item-stub activeclass="" tag="div" value="publication">
              <v-list-item-content-stub>
                <v-list-item-title-stub>
                  visualizations.add.data.data_types.publication
                </v-list-item-title-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
            <v-list-item-stub activeclass="" tag="div" value="publisher">
              <v-list-item-content-stub>
                <v-list-item-title-stub>
                  visualizations.add.data.data_types.publisher
                </v-list-item-title-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
            <v-list-item-stub activeclass="" tag="div" value="conference">
              <v-list-item-content-stub>
                <v-list-item-title-stub>
                  visualizations.add.data.data_types.conference
                </v-list-item-title-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
          </v-list-item-group-stub>
        </v-list-stub>
      </v-col-stub>
      <v-col-stub cols="12" lg="7" tag="div">
        <visualization-data-selection-bar-chart-stub chartid="bar-chart" width="400" height="400" cssclasses="" plugins="" entitydata="[object Object]" entitytype="visualizations.add.data.data_types.record" timeperiods=""></visualization-data-selection-bar-chart-stub>
      </v-col-stub>
      <v-col-stub cols="12" lg="3" tag="div">
        <h4>visualizations.add.data.set_time_period</h4>
        <v-container-stub tag="div" fluid="true">
          <v-row-stub tag="div" align="center">
            <v-col-stub cols="3" tag="div">
              <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" backgroundcolor="" label="visualizations.add.data.period.from" loaderheight="2" clearicon="$clear" placeholder="visualizations.add.data.period.enter_year" type="number" min="1970" max="2019"></v-text-field-stub>
            </v-col-stub>
            <v-col-stub cols="3" tag="div">
              <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" backgroundcolor="" label="visualizations.add.data.period.until" loaderheight="2" clearicon="$clear" placeholder="visualizations.add.data.period.enter_year" type="number" min="1970" max="2019"></v-text-field-stub>
            </v-col-stub>
            <v-col-stub cols="auto" tag="div">
              visualizations.add.data.period.elements
            </v-col-stub>
            <v-col-stub cols="auto" tag="div">
              <v-btn-stub tag="button" activeclass="" icon="true" type="button">
                <v-icon-stub>mdi-trash-can-outline</v-icon-stub>
              </v-btn-stub>
            </v-col-stub>
          </v-row-stub>
          <v-row-stub tag="div">
            <v-col-stub tag="div">
              <v-btn-stub tag="button" activeclass="" type="button">visualizations.add.data.period.add</v-btn-stub>
            </v-col-stub>
          </v-row-stub>
        </v-container-stub>
      </v-col-stub>
    </v-row-stub>
    <v-container-stub tag="div">
      <v-row-stub tag="div" justify="space-between" class="my-10">
        <v-col-stub cols="auto" tag="div">
          <v-btn-stub color="warning" tag="button" activeclass="" large="true" rounded="true" type="button">
            <v-icon-stub>mdi-menu-left</v-icon-stub>
            visualizations.add.data.back
          </v-btn-stub>
        </v-col-stub>
        <v-col-stub cols="auto" tag="div">
          <v-btn-stub color="success" tag="button" activeclass="" large="true" rounded="true" type="submit">visualizations.add.data.next
          </v-btn-stub>
        </v-col-stub>
      </v-row-stub>
    </v-container-stub>
  </v-container-stub>
</div>
`;

exports[`components/visualizations/edit/ChooseData.vue it handles the time periods can remove a time period 2`] = `
<div class="visualization-add-data" name="">
  <h1>visualizations.add.data.title</h1>
  <v-container-stub tag="div" fluid="true">
    <v-row-stub tag="div">
      <v-col-stub cols="12" lg="2" tag="div">
        <h4>visualizations.add.data.choose_type</h4>
        <v-list-stub tag="div" tile="true">
          <v-list-item-group-stub value="record" activeclass="v-item--active" mandatory="true" color="primary">
            <v-list-item-stub activeclass="" tag="div" value="record">
              <v-list-item-content-stub>
                <v-list-item-title-stub>
                  visualizations.add.data.data_types.record
                </v-list-item-title-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
            <v-list-item-stub activeclass="" tag="div" value="author">
              <v-list-item-content-stub>
                <v-list-item-title-stub>
                  visualizations.add.data.data_types.author
                </v-list-item-title-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
            <v-list-item-stub activeclass="" tag="div" value="institution">
              <v-list-item-content-stub>
                <v-list-item-title-stub>
                  visualizations.add.data.data_types.institution
                </v-list-item-title-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
            <v-list-item-stub activeclass="" tag="div" value="country">
              <v-list-item-content-stub>
                <v-list-item-title-stub>
                  visualizations.add.data.data_types.country
                </v-list-item-title-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
            <v-list-item-stub activeclass="" tag="div" value="publication">
              <v-list-item-content-stub>
                <v-list-item-title-stub>
                  visualizations.add.data.data_types.publication
                </v-list-item-title-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
            <v-list-item-stub activeclass="" tag="div" value="publisher">
              <v-list-item-content-stub>
                <v-list-item-title-stub>
                  visualizations.add.data.data_types.publisher
                </v-list-item-title-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
            <v-list-item-stub activeclass="" tag="div" value="conference">
              <v-list-item-content-stub>
                <v-list-item-title-stub>
                  visualizations.add.data.data_types.conference
                </v-list-item-title-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
          </v-list-item-group-stub>
        </v-list-stub>
      </v-col-stub>
      <v-col-stub cols="12" lg="7" tag="div">
        <visualization-data-selection-bar-chart-stub chartid="bar-chart" width="400" height="400" cssclasses="" plugins="" entitydata="[object Object]" entitytype="visualizations.add.data.data_types.record" timeperiods=""></visualization-data-selection-bar-chart-stub>
      </v-col-stub>
      <v-col-stub cols="12" lg="3" tag="div">
        <h4>visualizations.add.data.set_time_period</h4>
        <v-container-stub tag="div" fluid="true">
          <v-row-stub tag="div">
            <v-col-stub tag="div">
              <v-btn-stub tag="button" activeclass="" type="button">visualizations.add.data.period.add</v-btn-stub>
            </v-col-stub>
          </v-row-stub>
        </v-container-stub>
      </v-col-stub>
    </v-row-stub>
    <v-container-stub tag="div">
      <v-row-stub tag="div" justify="space-between" class="my-10">
        <v-col-stub cols="auto" tag="div">
          <v-btn-stub color="warning" tag="button" activeclass="" large="true" rounded="true" type="button">
            <v-icon-stub>mdi-menu-left</v-icon-stub>
            visualizations.add.data.back
          </v-btn-stub>
        </v-col-stub>
        <v-col-stub cols="auto" tag="div">
          <v-btn-stub color="success" tag="button" activeclass="" large="true" rounded="true" type="submit">visualizations.add.data.next
          </v-btn-stub>
        </v-col-stub>
      </v-row-stub>
    </v-container-stub>
  </v-container-stub>
</div>
`;

exports[`components/visualizations/edit/ChooseData.vue renders 1`] = `
<div class="visualization-add-data" name="">
  <h1>visualizations.add.data.title</h1>
  <v-container-stub tag="div" fluid="true">
    <v-row-stub tag="div">
      <v-col-stub cols="12" lg="2" tag="div">
        <h4>visualizations.add.data.choose_type</h4>
        <v-list-stub tag="div" tile="true">
          <v-list-item-group-stub value="record" activeclass="v-item--active" mandatory="true" color="primary">
            <v-list-item-stub activeclass="" tag="div" value="record">
              <v-list-item-content-stub>
                <v-list-item-title-stub>
                  visualizations.add.data.data_types.record
                </v-list-item-title-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
            <v-list-item-stub activeclass="" tag="div" value="author">
              <v-list-item-content-stub>
                <v-list-item-title-stub>
                  visualizations.add.data.data_types.author
                </v-list-item-title-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
            <v-list-item-stub activeclass="" tag="div" value="institution">
              <v-list-item-content-stub>
                <v-list-item-title-stub>
                  visualizations.add.data.data_types.institution
                </v-list-item-title-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
            <v-list-item-stub activeclass="" tag="div" value="country">
              <v-list-item-content-stub>
                <v-list-item-title-stub>
                  visualizations.add.data.data_types.country
                </v-list-item-title-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
            <v-list-item-stub activeclass="" tag="div" value="publication">
              <v-list-item-content-stub>
                <v-list-item-title-stub>
                  visualizations.add.data.data_types.publication
                </v-list-item-title-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
            <v-list-item-stub activeclass="" tag="div" value="publisher">
              <v-list-item-content-stub>
                <v-list-item-title-stub>
                  visualizations.add.data.data_types.publisher
                </v-list-item-title-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
            <v-list-item-stub activeclass="" tag="div" value="conference">
              <v-list-item-content-stub>
                <v-list-item-title-stub>
                  visualizations.add.data.data_types.conference
                </v-list-item-title-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
          </v-list-item-group-stub>
        </v-list-stub>
      </v-col-stub>
      <v-col-stub cols="12" lg="7" tag="div">
        <visualization-data-selection-bar-chart-stub chartid="bar-chart" width="400" height="400" cssclasses="" plugins="" entitydata="[object Object]" entitytype="visualizations.add.data.data_types.record" timeperiods=""></visualization-data-selection-bar-chart-stub>
      </v-col-stub>
      <v-col-stub cols="12" lg="3" tag="div">
        <h4>visualizations.add.data.set_time_period</h4>
        <v-container-stub tag="div" fluid="true">
          <v-row-stub tag="div">
            <v-col-stub tag="div">
              <v-btn-stub tag="button" activeclass="" type="button">visualizations.add.data.period.add</v-btn-stub>
            </v-col-stub>
          </v-row-stub>
        </v-container-stub>
      </v-col-stub>
    </v-row-stub>
    <v-container-stub tag="div">
      <v-row-stub tag="div" justify="space-between" class="my-10">
        <v-col-stub cols="auto" tag="div">
          <v-btn-stub color="warning" tag="button" activeclass="" large="true" rounded="true" type="button">
            <v-icon-stub>mdi-menu-left</v-icon-stub>
            visualizations.add.data.back
          </v-btn-stub>
        </v-col-stub>
        <v-col-stub cols="auto" tag="div">
          <v-btn-stub color="success" tag="button" activeclass="" large="true" rounded="true" type="submit">visualizations.add.data.next
          </v-btn-stub>
        </v-col-stub>
      </v-row-stub>
    </v-container-stub>
  </v-container-stub>
</div>
`;
